### ISSUE TITLE
Document LogPublisher.msg arguments

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6690]
|<img alt="zigg_'s avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| zigg_ reported|
|-|-|
|Trac ID|trac#6690|
|Type|defect|
|Created|2013-08-21 13:35:12Z|
|Branch|https://github.com/twisted/twisted/tree/log-msg-docs-6690|

The docstring for `LogPublisher.msg` does not document the keyword arguments that can be passed in.

Attachments:

* [LogPublisher_msg_documentation.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/711/711b6713b6386cdef1e87501156dda466eed2f6a/b8e0b542daf806133f7f336b8ba3c0dabddaa813.patch) (609 bytes) - added by zigg_ on 2013-08-21 13:35:46Z - patch to add documentation to LogPublisher.msg
* [LogPublisher_msg_documentation.2.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/711/711b6713b6386cdef1e87501156dda466eed2f6a/42828d7813e741460eebb0af6f434ec09c65e406.patch) (861 bytes) - added by zigg_ on 2013-08-21 15:30:42Z - Revised documentation for LogPublisher.msg
* [LogPublisher_msg_documentation.3.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/711/711b6713b6386cdef1e87501156dda466eed2f6a/d09c817b667cfe05b1c437cf0e7679beac6b6d8b.patch) (862 bytes) - added by zigg_ on 2013-09-24 18:25:06Z - document keyword arguments to LogPublisher.msg (replaces other patches)

<details><summary>Searchable metadata</summary>

```
trac-id__6690 6690
type__defect defect
reporter__zigg_ zigg_
priority__normal normal
milestone__ 
branch__branches_log_msg_docs_6690 branches/log-msg-docs-6690
branch_author__zigg_ zigg_
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1377092112000000 1377092112000000
changetime__1388261760000000 1388261760000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|<img alt="zigg_'s avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_2"></a>|zigg_ commented|
|-|-|

I've made substantial revisions based on feedback on IRC.  Thanks.

**Comment by twisted-trac** [comment]
|[<img alt="lvh's avatar" src="https://avatars.githubusercontent.com/u/97816?s=50" width="50" height="50">](https://github.com/lvh)<a name="note_3"></a>|@lvh set owner to zigg_|
|-|-|

Hi _zigg,


Thanks for working on this :-)

I'm assuming that I have to cut and paste from both diffs, since the former diff contains the argument documentation but the latter contains more usage documentation? I think both are important, and both should be added.

As a purely Python terminology comment, I think it's appropriate to change the first line of your usage docs:

```
Keyword arguments will be converted into keys in the event 
```

to:

```
Keyword arguments will be converted into items in the event 
```

... since both the key and value are used as a pair.

That's a minor nitpick though :)

Please resumbit with above notes (as one patch, please) and I'd be happy to review/run buildbots/merge.

cheers
lvh

**Comment by twisted-trac** [comment]
|<img alt="zigg_'s avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_4"></a>|zigg_ commented|
|-|-|

Actually, the second patch was intended to replace the first in its entirety.  The discussion on IRC came to the conclusion that the @params markup was improper to use with kw—better to explain it in prose—and message is already explained in prose.

I have gone ahead and generated a new patch with your little nitpick though. :-)

**Comment by twisted-trac** [comment]
|<img alt="zigg_'s avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|zigg_ removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun commented|
|-|-|

(In [41024]) Branching to 'log-msg-docs-6690'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun commented|
|-|-|

(In [41025]) apply LogPublisher_msg_documentation.3.patch

refs [#6690](https://github.com/twisted/twisted/issues/6690)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun commented|
|-|-|

Thanks.  Looks good to me.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_9"></a>|@exarkun set status to `closed`|
|-|-|

(In [41027]) Merge log-msg-docs-6690

Author: zigg_
Reviewer: exarkun
Fixes: [#6690](https://github.com/twisted/twisted/issues/6690)

Update the API documentation for `twisted.python.log.LogPublisher.msg` to
mention that keyword arguments are supported as well.
